src/actions/lendingActions.js

```diff
@@ -0,0 +1,196 @@
+      assets: { supportedAssets },
+      accounts: { data: accounts },
+      lending: { isFetchingDepositedAssets },
+    } = getState();
+
+    const currentAccountAssets = accountAssetsSelector(getState());
+    const smartWalletAccount = findFirstSmartAccount(accounts);
+    if (!smartWalletAccount) return;
+
+    if (isFetchingDepositedAssets) return;
+    dispatch({ type: SET_FETCHING_LENDING_DEPOSITED_ASSETS });
+
+    const depositedAssets = await aaveService.getAccountDepositedAssets(
+      getAccountAddress(smartWalletAccount),
+      getAssetsAsList(currentAccountAssets),
+      supportedAssets,
+    );
+    dispatch(setDepositedAssetsAction(depositedAssets));
+  };
+};
+
+export const fetchDepositedAssetAction = (symbol: string) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const {
+      assets: { supportedAssets },
+      accounts: { data: accounts },
+      lending: { depositedAssets, isFetchingDepositedAssets },
+    } = getState();
+    const currentAccountAssets = accountAssetsSelector(getState());
+    const smartWalletAccount = findFirstSmartAccount(accounts);
+    if (!smartWalletAccount) return;
+
+    if (isFetchingDepositedAssets) return;
+    dispatch({ type: SET_FETCHING_LENDING_DEPOSITED_ASSETS });
+
+    const asset = getAssetData(getAssetsAsList(currentAccountAssets), supportedAssets, symbol);
+    const accountAddress = getAccountAddress(smartWalletAccount);
+
+    const updatedDepositedAsset = await aaveService.fetchAccountDepositedAsset(accountAddress, asset);
+    if (!updatedDepositedAsset) return;
+
+    const updatedDepositedAssets = depositedAssets.reduce((
+      currentList,
+      depositedAsset,
+      depositedAssetIndex,
+    ) => {
+      if (updatedDepositedAsset.symbol === depositedAsset.symbol) {
+        currentList[depositedAssetIndex] = updatedDepositedAsset;
+      }
+      return currentList;
+    }, depositedAssets);
+
+    dispatch(setDepositedAssetsAction(updatedDepositedAssets));
+  };
+};
+
+export const calculateLendingDepositTransactionEstimateAction = (
+  amount: number,
+  asset: AssetToDeposit,
+) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const { accounts: { data: accounts } } = getState();
+    const smartWalletAccount = findFirstSmartAccount(accounts);
+    if (!smartWalletAccount) return;
+
+    dispatch({ type: SET_CALCULATING_LENDING_DEPOSIT_TRANSACTION_ESTIMATE });
+
+    // may include approve transaction
+    const aaveDepositNeededTransactions = await getAaveDepositTransactions(
+      getAccountAddress(smartWalletAccount),
+      amount,
+      asset,
+    );
+
+    const estimateTransactions = aaveDepositNeededTransactions.map(({
+      to: recipient,
+      amount: value,
+      data,
+    }) => ({ recipient, value, data }));
+
+    const estimateTransaction: AccountTransaction = {
+      ...estimateTransactions[0],
+      sequentialTransactions: estimateTransactions.slice(1), // exclude first, take rest if exist
```

Comment:
Keep in mind we support max 5 tx per batch

Reply:
Yes, there's a comment for that in Smart Wallet service under estimate and submit methods.